src/OpenCensus/Resource/Resource.cs

```diff
@@ -0,0 +1,159 @@
+
+        /// <summary>
+        /// Tag list splitter.
+        /// </summary>
+        private const char LABEL_LIST_SPLITTER = ',';
+
+        /// <summary>
+        /// Key-value splitter.
+        /// </summary>
+        private const char LABEL_KEY_VALUE_SPLITTER = '=';
+
+        /// <summary>
+        /// Error message when string contains invalid characters.
+        /// </summary>
+        private static readonly string ERROR_MESSAGE_INVALID_CHARS =
+            " should be a ASCII string with a length greater than 0 and not exceed " + MaxLength + " characters.";
+
+        /// <summary>
+        /// Environment identification (for example, AKS/GKE/etc).
+        /// </summary>
+        private static readonly string EnvironmentType = ParseResourceType(Environment.GetEnvironmentVariable(OC_RESOURCE_TYPE_ENV));
+
+        private static readonly ITag[] EnvironmentToLabelMap =
+            ParseResourceLabels(Environment.GetEnvironmentVariable(OC_RESOURCE_LABELS_ENV));
+
+        /// <summary>
+        /// Gets the identification of the resource.
+        /// </summary>
+        public abstract string Type { get; protected set; }
+
+        /// <summary>
+        /// Gets the map between the tag and its value.
+        /// </summary>
+        public abstract IList<ITag> Tags { get; }
+
+        /// <summary>
+        /// Creates a label/tag map from the OC_RESOURCE_LABELS environment variable.
+        /// OC_RESOURCE_LABELS: A comma-separated list of labels describing the source in more detail,
+        /// e.g. “key1=val1,key2=val2”. Domain names and paths are accepted as label keys.
+        /// Values may be quoted or unquoted in general. If a value contains whitespaces, =, or " characters, it must
+        /// always be quoted.
+        /// </summary>
+        /// <param name="rawEnvironmentTags">Environment tags as a raw, comma separated string</param>
+        /// <returns>Environment Tags as a list</returns>
+        private static ITag[] ParseResourceLabels(string rawEnvironmentTags)
+        {
+            if (rawEnvironmentTags == null)
+            {
+                return new ITag[0] { };
+            }
+            else
+            {
+                var labels = new List<ITag>();
+                string[] rawLabels = rawEnvironmentTags.Split(LABEL_LIST_SPLITTER);
+                foreach (var rawLabel in rawLabels)
+                {
+                    string[] keyValuePair = rawLabel.Split(LABEL_KEY_VALUE_SPLITTER);
+                    if (keyValuePair.Length != 2)
+                    {
+                        continue;
+                    }
+
+                    string key = keyValuePair[0].Trim();
+                    string value = Regex.Replace(keyValuePair[1].Trim(), "^\"|\"$", string.Empty);
+
+                    Arguments.Check(IsValidAndNotEmpty(key), "Label key" + ERROR_MESSAGE_INVALID_CHARS);
+                    Arguments.Check(IsValid(value), "Label value" + ERROR_MESSAGE_INVALID_CHARS);
```

Comment:
Again, throwing exception is not optimal here. Perhaps error logging and ignoring of the string would work better